### PR TITLE
NXCM-3661, allow all CRUDS for non exposed repositories ONLY over service/local

### PR DESCRIPTION
**Pull request by bdemers**

### CODE REVIEW

File: nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/rest/AbstractResourceStoreContentPlexusResource.java [hunk 1]
```diff
@@ -258,20 +258,7 @@ protected ResourceStore getResourceStore( final Request request, final Action ac
...
-        if ( !action.isReadAction() && resourceStore instanceof Repository )
-        {
-            if ( !( (Repository) resourceStore ).isExposed() )
-            {
-                throw new NoSuchRepositoryException( ( (Repository) resourceStore ).getId() );
-            }
-        }
-
-        return resourceStore;
+        return getResourceStore( request );
```
Comment (cstamas) [inline nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/rest/AbstractResourceStoreContentPlexusResource.java]: This method was introduced as part of NXCM-3600. As of this change, the existence of the method lost it's meaning, should be removed. See here:
https://github.com/sonatype/nexus-enterprise/blob/master/nexus-enterprise-plugins/nexus-staging-plugin/src/main/java/com/sonatype/nexus/staging/api/StagingPlexusResource.java#L89

**Comment by velo** [comment]
+1

**Comment by bdemers** [comment]
CI +1 (sort of)
Linux:
https://builds.sonatype.org/view/nexus/job/nexus-oss-its-feature/jdk=java-6x,label=linux/166/testReport/ reports a test failure, but I ran the nexus-proxy module again with sucess
https://builds.sonatype.org/view/nexus/job/nexus-oss-its-feature/jdk=java-6x,label=linux/167/testReport/

Windows passed the first time
https://builds.sonatype.org/view/nexus/job/nexus-oss-its-feature/jdk=java-6x,label=win/166/testReport/
Then the nexus-proxy module failed
https://builds.sonatype.org/view/nexus/job/nexus-oss-its-feature/jdk=java-6x,label=win/167/testReport/

NOTE: there are no nexus-proxy changes in this pull

**Comment by peterlynch** [comment]
+1

**Comment by mcculls** [comment]
+1 looks reasonable

**Pull request merged**